From da11533f66c1724a9b295c0e707d0645051dbd1f Mon Sep 17 00:00:00 2001
From: AsavarTzeth <asavartzeth@gmail.com>
Date: Thu, 1 Mar 2018 10:33:22 +0100
Subject: [PATCH 1/4] Add flatpak distribution type

Enable support for the CMake option: KEEPASSXC_DIST_TYPE=Flatpak.

This can be used to install certain files with reverse-dns style names
as well as making other minor changes without affecting any other
distribution type.

It will also show Flatpak as the distribution type in the application
gui, so that bug reports correctly reflect this.
---
 CMakeLists.txt              | 5 ++++-
 src/config-keepassx.h.cmake | 1 +
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16d574f0..13624462 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@
 cmake_minimum_required(VERSION 3.1.0)
 
 project(KeePassXC)
+set(ID "org.keepassxc.${PROJECT_NAME}")
 
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING
@@ -149,11 +150,13 @@ message(STATUS "Setting up build for KeePassXC v${KEEPASSXC_VERSION}\n")
 # Distribution info
 set(KEEPASSXC_DIST ON)
 set(KEEPASSXC_DIST_TYPE "Other" CACHE STRING "KeePassXC Distribution Type")
-set_property(CACHE KEEPASSXC_DIST_TYPE PROPERTY STRINGS Snap AppImage Other)
+set_property(CACHE KEEPASSXC_DIST_TYPE PROPERTY STRINGS Snap AppImage Flatpak Other)
 if(KEEPASSXC_DIST_TYPE STREQUAL "Snap")
     set(KEEPASSXC_DIST_SNAP ON)
 elseif(KEEPASSXC_DIST_TYPE STREQUAL "AppImage")
     set(KEEPASSXC_DIST_APPIMAGE ON)
+elseif(KEEPASSXC_DIST_TYPE STREQUAL "Flatpak")
+    set(KEEPASSXC_DIST_FLATPAK ON)
 elseif(KEEPASSXC_DIST_TYPE STREQUAL "Other")
     unset(KEEPASSXC_DIST)
 endif()
diff --git a/src/config-keepassx.h.cmake b/src/config-keepassx.h.cmake
index 2acff446..134b948b 100644
--- a/src/config-keepassx.h.cmake
+++ b/src/config-keepassx.h.cmake
@@ -32,6 +32,7 @@
 #cmakedefine KEEPASSXC_DIST_TYPE "@KEEPASSXC_DIST_TYPE@"
 #cmakedefine KEEPASSXC_DIST_SNAP
 #cmakedefine KEEPASSXC_DIST_APPIMAGE
+#cmakedefine KEEPASSXC_DIST_FLATPAK
 
 #cmakedefine HAVE_PR_SET_DUMPABLE 1
 #cmakedefine HAVE_RLIMIT_CORE 1
-- 
2.21.0

